### PR TITLE
[QC-837] Apply policies only for data in a given period

### PR DESCRIPTION
**Pull request by Barthelemy**
One can now add some options in the config file to apply rules only in a given time period. 
I also added a number of tests.

### CODE REVIEW

**Comment by Barthelemy** [comment]
I WIP it because I don't want it to be installed at P2 while I am absent.

**Comment by Barthelemy** [comment]
@knopers8 it is ready for review

File: Framework/script/RepoCleaner/rules/last_only.py [hunk 1]
```diff
@@ -34,24 +36,32 @@ def process(ccdb: Ccdb, object_path: str, delay: int, extra_params: Dict[str, st
         if earliest == None or v.validFromAsDt > earliest.validFromAsDt:
             earliest = v
     logger.debug(f"earliest : {earliest}")
-            
+
+    logger.debug(f"{from_timestamp}")
+    logger.debug(f"{to_timestamp}")
```
Comment (knopers8) [inline Framework/script/RepoCleaner/rules/last_only.py:41]: if you want to keep it, then please explain what will the two numbers mean in the logs.

**Review by knopers8 — APPROVED** [review]
Thank you, it is good as it is, but please consider some pedantic suggestions.

File: Framework/script/RepoCleaner/rules/1_per_hour.py [hunk 2]
```diff
@@ -40,25 +41,28 @@ def process(ccdb: Ccdb, object_path: str, delay: int, #migration: bool,
...
-                deletion_list.append(v)
-                logger.debug(f"not in the grace period, we delete {v}")
-                ccdb.deleteVersion(v)
-            else:
-                preservation_list.append(v)
-
-    logger.debug("deleted : ")
+            if v.validFromAsDt < datetime.now() - timedelta(minutes=delay):  # grace period
+                logger.debug(f"not in the grace period")
+                if from_timestamp < v.validFrom < to_timestamp:  # in the allowed period
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/script/RepoCleaner/rules/1_per_hour.py:50]: https://www.geeksforgeeks.org/chaining-comparison-operators-python/

File: Framework/script/RepoCleaner/o2-qc-repo-cleaner [hunk 3]
```diff
@@ -115,9 +119,19 @@ def parseConfig(config_file_path):
...
-        rule = Rule(rule_yaml["object_path"], rule_yaml["delay"], rule_yaml["policy"],  # rule_yaml["migration"],
-                    rule_yaml)
+        if "from_timestamp" in rule_yaml:
+            from_timestamp = rule_yaml["from_timestamp"]
+        else:
+            from_timestamp = 1
+        if "to_timestamp" in rule_yaml:
+            to_timestamp = rule_yaml["to_timestamp"]
+        else:
+            to_timestamp = 1975323342000  # 2033
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/script/RepoCleaner/o2-qc-repo-cleaner:131]: I have put your retirement date now, it is large enough ;)

File: Framework/script/RepoCleaner/rules/last_only.py (hunk 1 above)
Comment (Barthelemy) [reply Framework/script/RepoCleaner/rules/last_only.py:41]: removed

**Pull request merged**